### PR TITLE
Bug 1836357 - Add a dialog for handling FedCM showProviderPrompt.

### PR DESCRIPTION
**Pull request by Amejia481**
Depends on https://bugzilla.mozilla.org/show_bug.cgi?id=1836356

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1836357

### CODE REVIEW

File: android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt [hunk 1]
```diff
@@ -1183,6 +1184,59 @@ class GeckoPromptDelegateTest {
...
+
+            // Just making sure we are not completing the geckoResult twice.
+            onConfirmWasCalled = false
+            onConfirm(acProvider)
+            shadowOf(getMainLooper()).idle()
+            assertFalse(onConfirmWasCalled)
+        }
+
+        // Verifying we are handling the dismiss correctly.
+        geckoResult = promptDelegate.onSelectIdentityCredentialProvider(mock(), geckoProviderSelectorPrompt(listOf(geckoProvider)))
```
Comment (Amejia481) [inline android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt:1230]: @titooan I figured out what was the issue, I was reusing the `geckoResult`, which was already completed, now I'm just calling  `geckoProviderSelectorPrompt` again.

**Comment by Amejia481** [comment]
Already [Bug #1836356](https://bugzilla.mozilla.org/show_bug.cgi?id=1836356) landed, rebasing on top of the latest GV version.

**Comment by Amejia481** [comment]
The tasks failures are unrelated to our changes, maybe a bad build.

```
java.lang.UnsatisfiedLinkError: Error looking up function 'ffi_glean_64d5_OnGleanEvents_init_callback': undefined symbol: ffi_glean_64d5_OnGleanEvents_init_callback

     FATAL EXCEPTION: main
Process: org.mozilla.samples.browser, PID: 7232
java.lang.UnsatisfiedLinkError: Error looking up function 'ffi_glean_64d5_OnGleanEvents_init_callback': undefined symbol: ffi_glean_64d5_OnGleanEvents_init_callback
	at com.sun.jna.Function.<init>(Function.java:252)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:620)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:596)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:582)
	at com.sun.jna.Library$Handler.invoke(Library.java:248)
	at java.lang.reflect.Proxy.invoke(Proxy.java:1006)
	at $Proxy4.ffi_glean_64d5_OnGleanEvents_init_callback(Unknown Source)
	at mozilla.telemetry.glean.internal.FfiConverterTypeOnGleanEvents.register$glean_release(glean.kt:4120)
	at mozilla.telemetry.glean.internal._UniFFILib$Companion$INSTANCE$2.invoke(glean.kt:261)
	at mozilla.telemetry.glean.internal._UniFFILib$Companion$INSTANCE$2.invoke(glean.kt:258)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at mozilla.telemetry.glean.internal._UniFFILib$Companion.getINSTANCE$glean_release(glean.kt:258)
	at mozilla.telemetry.glean.internal.GleanKt.gleanEnableLogging(glean.kt:4766)
	at mozilla.telemetry.glean.GleanInternalAPI.<init>(Glean.kt:142)
	at mozilla.telemetry.glean.Glean.<init>(Glean.kt:641)
	at mozilla.telemetry.glean.Glean.<clinit>(Unknown Source:2)
	at mozilla.components.service.glean.Glean.initialize(Glean.kt:49)
	at org.mozilla.samples.browser.SampleApplication.onCreate(SampleApplication.kt:72)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
	at androidx.test.runner.MonitoringInstrumentation.callApplicationOnCreate(MonitoringInstrumentation.java:483)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
	at android.app.ActivityThread.access$1100(ActivityThread.java:199)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

```

**Comment by Amejia481** [comment]
Failures are related to https://bugzilla.mozilla.org/show_bug.cgi?id=1837869#c8

File: android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt (hunk 1 above)
Comment (titooan) [reply android-components/components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegateTest.kt:1230]: Oh great, well done!

**Review by titooan — APPROVED** [review]
LGTM! 👏

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/identitycredential/Provider.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import android.annotation.SuppressLint
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Represents an Identity credential provider:
+ * @property id An identifier for this [Provider].
+ * @property icon an icon of the provider normally the logo of the brand.
+ * @property name The text name of the this [Provider].
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply android-components/components/concept/engine/src/main/java/mozilla/components/concept/identitycredential/Provider.kt:15]: Addressed!

File: android-components/components/concept/engine/src/main/java/mozilla/components/concept/identitycredential/Provider.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+package mozilla.components.concept.identitycredential
+
+import android.annotation.SuppressLint
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Represents an Identity credential provider:
+ * @property id An identifier for this [Provider].
+ * @property icon an icon of the provider normally the logo of the brand.
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply android-components/components/concept/engine/src/main/java/mozilla/components/concept/identitycredential/Provider.kt:14]: Addressed!

**Pull request merged**